client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts

```diff
@@ -608,32 +568,74 @@ const siteMigration: FlowV2 = {
 						return exitFlow( `/home/${ siteId }` );
 					}
 
-					return navigate( `${ STEPS.SITE_MIGRATION_HOW_TO_MIGRATE.slug }?${ urlQueryParams }` );
+					return navigate(
+						paths.howToMigratePath( {
+							siteSlug,
+							siteId,
+							from: fromQueryParam,
+						} )
+					);
 				}
 
 				case STEPS.SITE_MIGRATION_OTHER_PLATFORM_DETECTED_IMPORT.slug: {
-					return navigate( `${ STEPS.SITE_MIGRATION_CREDENTIALS.slug }?${ urlQueryParams }` );
+					return navigate(
+						paths.credentialsPath( {
+							siteId,
+							siteSlug,
+							from: fromQueryParam,
+						} )
+					);
 				}
 
 				case STEPS.SITE_MIGRATION_FALLBACK_CREDENTIALS.slug: {
 					if (
 						urlQueryParams.get( 'backTo' ) ===
 						STEPS.SITE_MIGRATION_APPLICATION_PASSWORD_AUTHORIZATION.slug
 					) {
+						const queryParams = Object.fromEntries( urlQueryParams );
 						return navigate(
-							`${ STEPS.SITE_MIGRATION_APPLICATION_PASSWORD_AUTHORIZATION.slug }?${ urlQueryParams }`
+							paths.applicationPasswordAuthorizationPath( {
+								siteId,
+								siteSlug,
+								from: fromQueryParam,
+								...queryParams,
```

Comment:
Ideally, we should remove this generic param and use strictly what we need ; I plan to do it in a further interaction